This is the dataset of Nepal Census 2021..


In [1]:
import numpy as np
import pandas as pd

In [2]:
data=pd.read_csv("census.csv")

In [3]:
data

,_id,District,Local Level Name,Total family number,Total household number,Total population,Total Male,Total Female
0,1,Taplejung,Aathrai Tribeni Rural Municipality,2869,2735,12288,6005,6283
1,2,Taplejung,Maiwakhola Rural Municipality,2275,2178,10365,5264,5101
2,3,Taplejung,Meringden Rural Municipality,2683,2528,12040,6181,5859
3,4,Taplejung,Mikwakhola Rural Municipality,1862,1792,7991,4000,3991
4,5,Taplejung,Phaktanglung Rural Municipality,2864,2700,11925,6239,5686
...,...,...,...,...,...,...,...,...
748,749,Darchula,Mahakali Municipality,6080,4596,24572,12072,12500
749,750,Darchula,Malikaarjun Rural Municipality,3241,2993,15754,7631,8123
750,751,Darchula,Marma Rural Municipality,3086,2704,15586,7551,8035
751,752,Darchula,Naugad Rural Municipality,3156,2920,16434,7988,8446


Data Understanding (5 questions)
1. What are the shape (rows & columns), data types, and basic info of the dataset?

In [4]:
data.shape

(753, 8)

our data set have 753 rows and 8 columns

In [5]:
data.columns

Index(['_id', 'District', 'Local Level Name', 'Total family number',
       'Total household number', 'Total population', 'Total Male',
       'Total Female'],
      dtype='object')

these are our columns:

id = id number 

District = Name of the district

Local Level Name = name of the municipality under the district

Total family number = Total number of families

Total household number = total number of houses 

Total population = total number of people

Total Male = total number of male population

Total Female = Total number of female population

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 753 entries, 0 to 752
Data columns (total 8 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   _id                     753 non-null    int64 
 1   District                753 non-null    object
 2   Local Level Name        753 non-null    object
 3   Total family number     753 non-null    int64 
 4   Total household number  753 non-null    int64 
 5   Total population        753 non-null    int64 
 6   Total Male              753 non-null    int64 
 7   Total Female            753 non-null    int64 
dtypes: int64(6), object(2)
memory usage: 47.2+ KB


2. How many unique Districts and Local Levels are there?

In [17]:
data['District'].unique()

array(['Taplejung', 'Panchthar', 'Ilam', 'Jhapa', 'Morang', 'Sunsari',
       'Dhankuta', 'Terhathum', 'Sankhuwasabha', 'Bhojpur', 'Solukhumbu',
       'Okhaldhunga', 'Khotang', 'Udayapur', 'Saptari', 'Siraha',
       'Dhanusa', 'Mahottari', 'Sarlahi', 'Rautahat', 'Bara', 'Parsa',
       'Sindhuli', 'Ramechhap', 'Dolakha', 'Sindhupalchok',
       'Kavrepalanchok', 'Lalitpur', 'Bhaktapur', 'Kathmandu', 'Nuwakot',
       'Rasuwa', 'Dhading', 'Makwanpur', 'Chitawan', 'Gorkha', 'Lamjung',
       'Tanahu', 'Syangja', 'Kaski', 'Manang', 'Mustang', 'Myagdi',
       'Parbat', 'Baglung', 'Nawalparasi East', 'Gulmi', 'Palpa',
       'Nawalparasi West', 'Rupandehi', 'Kapilbastu', 'Arghakhanchi',
       'Pyuthan', 'Rolpa', 'Rukum East', 'Dang', 'Banke', 'Bardiya',
       'Rukum West', 'Salyan', 'Surkhet', 'Dailekh', 'Jajarkot', 'Dolpa',
       'Jumla', 'Kalikot', 'Mugu', 'Humla', 'Bajura', 'Bajhang', 'Achham',
       'Doti', 'Kailali', 'Kanchanpur', 'Dadeldhura', 'Baitadi',
       'Darchula'], dty

In [15]:
len(data['District'].unique())


77

77 unique districts

In [23]:
data["Local Level Name"].unique()

array(['Aathrai Tribeni Rural Municipality',
       'Maiwakhola Rural Municipality', 'Meringden Rural Municipality',
       'Mikwakhola Rural Municipality', 'Phaktanglung Rural Municipality',
       'Phungling Municipality', 'Sidingba Rural Municipality',
       'Sirijangha Rural Municipality',
       'Pathivara Yangwarak Rural Municipality',
       'Falelung Rural Municipality', 'Falgunanda Rural Municipality',
       'Hilihang Rural Municipality', 'Kummayak Rural Municipality',
       'Miklajung Rural Municipality', 'Phidim Municipality',
       'Tumbewa Rural Municipality', 'Yangwarak Rural Municipality',
       'Chulachuli Rural Municipality', 'Deumai Municipality',
       'Fakphokthum Rural Municipality', 'Illam Municipality',
       'Maijogmai Rural Municipality', 'Mai Municipality',
       'Mangsebung Rural Municipality', 'Rong Rural Municipality',
       'Sandakpur Rural Municipality', 'Suryodaya Municipality',
       'Arjundhara Municipality', 'Barhadashi Rural Municipality',


In [24]:
len(data['Local Level Name'].unique())

737

In [25]:
data['Local Level Name'].nunique()

737

737 unique Local level names.

3. Are there any missing values or duplicate rows in the dataset?

In [26]:
data.isnull().sum()

_id                       0
District                  0
Local Level Name          0
Total family number       0
Total household number    0
Total population          0
Total Male                0
Total Female              0
dtype: int64

No, there is no any missing values

In [27]:
data.duplicated().sum()

np.int64(0)

No, their are no duplicate rows.

4. What is the summary statistics (mean, median, min, max, std) for numerical columns like Total Population, Total Male, Total Female, etc.?

In [28]:
data.describe()

,_id,Total family number,Total household number,Total population,Total Male,Total Female
count,753.000000,753.000000,753.000000,753.000000,753.000000,753.000000
mean,377.000000,8976.723772,7493.590969,38612.204515,18842.814077,19769.390438
std,217.516666,12986.552238,8401.892705,49904.377212,24891.928559,25046.362147
min,1.000000,125.000000,125.000000,398.000000,170.000000,228.000000
25%,189.000000,4036.000000,3653.000000,16768.000000,8115.000000,8553.000000
50%,377.000000,5896.000000,5195.000000,25508.000000,12248.000000,13010.000000
75%,565.000000,9510.000000,8151.000000,44929.000000,22100.000000,23068.000000
max,753.000000,231714.000000,105649.000000,845767.000000,431501.000000,414266.000000


5. What is the range and distribution of Total Family Number and Total Household Number?

In [35]:
# to find range 
min_family_num = data['Total family number'].min()
max_family_num = data['Total family number'].max()
family_range = max_family_num - min_family_num
print(f"Range of Total family number = {family_range}")

min_household = data['Total household number'].min()
max_household = data['Total household number'].max()
household_range = max_household - min_household
print(f"Range of Total household number = {household_range}")

Range of Total family number = 231589
Range of Total household number = 105524


In [36]:
# to find distribution
data['Total family number'].describe()

count       753.000000
mean       8976.723772
std       12986.552238
min         125.000000
25%        4036.000000
50%        5896.000000
75%        9510.000000
max      231714.000000
Name: Total family number, dtype: float64

In [37]:
data["Total household number"].describe()

count       753.000000
mean       7493.590969
std        8401.892705
min         125.000000
25%        3653.000000
50%        5195.000000
75%        8151.000000
max      105649.000000
Name: Total household number, dtype: float64